src/batch/src/task/task_execution.rs

```diff
@@ -366,8 +373,17 @@ impl<C: BatchTaskContext> BatchTaskExecution<C> {
                     error!("Execution failed [{:?}]: {}", &task_id, e);
                     let err_str = e.to_string();
                     *failure.lock() = Some(e);
+                    // There will be no more chunks, so send None.
+                    if let Err(_e) = sender.send(None).await {
```

Comment:
I think we should not send a `None` here, otherwise, the upstream task can't  tell whether this task finished normally or failed. Also there is no guarantee that the execution failure sender happens before this sender if they are in different channels.

Reply:
I have consider it

1. Currently our channel can not send Chunk, can not send Error.
2. execution failure sender happens before this sender if they are in different channels. I don't think it will be a problem? Cuz I don't care which one happen early or late. executor A fails, it should only report this to frontend/upstream stage in local mode, and it's parent executor in the same task should also stopped, but not report the error.

I also do not want to add this, but if I not, there will be a stuck problem may be hard to debug🤔

Reply:
Oh I may have misunderstand 2. If you are talking about `local_exec_failure_sender` in CN, then I think it must be happen before send None? (.await)

Reply:
According to [this doc](https://docs.rs/tokio/latest/tokio/sync/mpsc/struct.Sender.html#method.send), I think the `.await` means that it has been put into channel rather than guaranteeing receiver has received it.

Reply:
This is fine. 

Because at least it send the error into the failure sender channel, as long as the receiver is not dead, it will always be processed. Even if the parent executor first receive the None, it will not be a problem? Because there indeed there will be no Chunk to send in and only None.

Reply:
I think I know why it stucks.

Reply:
To resolve the stuck:
1. Remove the call of `take_output_num`
2. The send `None` should be after the send local error failure.

Reply:
Yes I think this pr choose the 2

Reply:
No, both of them should happen

Reply:
When we only do 2, the effect is mostly the same as 1 + 2, and the change is minimal. 

But I agree that we can remove the call of `take_output_num` to make the logic more simple and clean.

Reply:
We haven't finish 2 yet, the error is still sent after `None`